project_task_id_in_display_name/models/project_task.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2024-today Numigi and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
+
+
+class ProjectTask(models.Model):
+    """
+    Add a field to allow searching a task by its ID.
```

Comment:
Hello @rivo2302 

Please make sure that this sentence is always true in V16
In vanilla project module, we can search by task ID 

[See code here](https://github.com/odoo/odoo/blob/16.0/addons/project/views/project_views.xml#L19)